packages/tools/src/utilities/boundingBox/getBoundingBoxAroundShape.ts

```diff
@@ -1,57 +1,62 @@
 import type { Types } from '@cornerstonejs/core';
 
 /**
- * With a given vertices (points) coordinates in IJK, it calculates the minimum and maximum
- * coordinate in each axis, and returns them. If dimensions are provided it also
+ * With a given vertices (points) coordinates in 2D or 3D IJK, it calculates the minimum and maximum
+ * coordinate in each axis, and returns them. If clipBounds are provided it also
  * clip the min, max to the provided width, height and depth
  *
  * @param points - shape corner points coordinates (IJK)
- * @param dimensions - dimensions of the image
+ * @param clipBounds - bounds to clip the min, max
  * @returns [[xMin,xMax],[yMin,yMax], [zMin,zMax]]
  */
 function getBoundingBoxAroundShape(
-  points: Types.Point3[],
-  dimensions?: Types.Point3
-): [Types.Point2, Types.Point2, Types.Point2] {
+  points: Types.Point2[] | Types.Point3[],
+  clipBounds?: Types.Point2 | Types.Point3
+): [Types.Point2, Types.Point2, Types.Point2 | null] {
   let xMin = Infinity;
-  let xMax = 0;
+  let xMax = -Infinity;
   let yMin = Infinity;
-  let yMax = 0;
+  let yMax = -Infinity;
   let zMin = Infinity;
-  let zMax = 0;
+  let zMax = -Infinity;
+
+  const is3D = points[0] && points[0].length === 3;
 
   points.forEach((p) => {
     xMin = Math.min(p[0], xMin);
     xMax = Math.max(p[0], xMax);
     yMin = Math.min(p[1], yMin);
     yMax = Math.max(p[1], yMax);
-    zMin = Math.min(p[2], zMin);
-    zMax = Math.max(p[2], zMax);
+
+    if (is3D) {
+      zMin = Math.min(p[2] ?? zMin, zMin);
+      zMax = Math.max(p[2] ?? zMax, zMax);
+    }
   });
 
-  xMin = Math.floor(xMin);
-  xMax = Math.floor(xMax);
-  yMin = Math.floor(yMin);
-  yMax = Math.floor(yMax);
-  zMin = Math.floor(zMin);
-  zMax = Math.floor(zMax);
+  // IMPORTANT: we used to do flooring here, which really
+  // did not make sense.
 
-  if (dimensions) {
-    // clip the min, max to the provided width, height and depth
-    const [width, height, depth] = dimensions;
+  if (clipBounds) {
+    // clip the min, max to the provided dimensions
     xMin = Math.max(0, xMin);
-    xMax = Math.min(width - 1, xMax);
+    xMax = Math.min(clipBounds[0] - 1, xMax);
```

Comment:
If these are float values, then you should clip to clipBounds[0]-EPS, and not clipBounds[0]-1 
That is, if you aren't using the floor up above, then anything less than clipBounds is ok